spatial/r2/vector_test.go

```diff
@@ -135,3 +138,46 @@ func TestCross(t *testing.T) {
 		})
 	}
 }
+
+func TestNorm(t *testing.T) {
+	for _, test := range []struct {
+		v    Vec
+		want float64
+	}{
+		{Vec{0, 0}, 0},
+		{Vec{0, 1}, 1},
+		{Vec{1, 1}, math.Sqrt2},
+		{Vec{1, 2}, math.Sqrt(5)},
+		{Vec{3, -4}, 5},
+		{Vec{1, 1e-16}, 1},
+		// This will overflow the float64, and should return zero.
```

Comment:
`// This will underflow and return zero.`

Although check this with the new code.